waku/waku.go

```diff
@@ -56,15 +55,18 @@ type Bridge interface {
 }
 
 type settings struct {
-	MaxMsgSize               uint32  // Maximal message length allowed by the waku node
-	EnableConfirmations      bool    // Enable sending message confirmations
-	MinPow                   float64 // Minimal PoW required by the waku node
-	MinPowTolerance          float64 // Minimal PoW tolerated by the waku node for a limited time
-	BloomFilter              []byte  // Bloom filter for topics of interest for this node
-	BloomFilterTolerance     []byte  // Bloom filter tolerated by the waku node for a limited time
-	LightClient              bool    // Light client mode enabled does not forward messages
-	RestrictLightClientsConn bool    // Restrict connection between two light clients
-	SyncAllowance            int     // Maximum time in seconds allowed to process the waku-related messages
+	MaxMsgSize               uint32             // Maximal message length allowed by the waku node
+	EnableConfirmations      bool               // Enable sending message confirmations
+	MinPow                   float64            // Minimal PoW required by the waku node
+	MinPowTolerance          float64            // Minimal PoW tolerated by the waku node for a limited time
+	BloomFilter              []byte             // Bloom filter for topics of interest for this node
+	BloomFilterTolerance     []byte             // Bloom filter tolerated by the waku node for a limited time
+	TopicInterest            map[TopicType]bool // Topic interest for this node
+	TopicInterestTolerance   map[TopicType]bool // Topic interest tolerated by the waku node for a limited time
+	BloomFilterMode          bool               // Whether we should match against bloom-filter only
+	LightClient              bool               // Light client mode enabled does not forward messages
+	RestrictLightClientsConn bool               // Restrict connection between two light clients
+	SyncAllowance            int                // Maximum time in seconds allowed to process the waku-related messages
```

Comment:
What is this for? I know it was there before but it isn't clear at a glance

Reply:
yes, makes sense, although I take is up to each client to decide how strict they want to be

Reply:
I wouldn't specify specific timeouts, it really depends on clients and there use cases. But I guess recommendations do no harm. We currently have something like this mentioned for timeout on the handshake